<launch>
    <!-- This is drone 1 -->
    <arg name="rosbag_path" default="/root/catkin_ws/src/offboard_pkg/bag/demo_experience_bigger_2025-01-08-20-57-50.bag"/>
    <arg name="rosbag_topic" default="/drone_1/mavros/setpoint_raw/local"/>
    <arg name="topic_start_seq" default="403"/>
    <arg name="topic_end_seq" default="1314"/>
    <!-- must same as offboard trajectory.launch -->
    <arg name="flight_height" default="1.5"/>
    <arg name="control_height" default="true"/>
    <!-- 启动offboard_waypoint_tracker节点 -->
    <node pkg="offboard_pkg" type="offboard_setpoint_raw_tracker" name="offboard_setpoint_raw_tracker_node" output="screen" >
        <!-- 使用传入的参数设置rosbag路径参数 -->
        <param name="rosbag_path" value="$(arg rosbag_path)" />
        <param name="rosbag_topic" value="$(arg rosbag_topic)" />
        <param name="topic_start_seq" value="$(arg topic_start_seq)" />
        <param name="topic_end_seq" value="$(arg topic_end_seq)"/>
        <param name="flight_height" value="$(arg flight_height)"/>
        <param name="control_height" value="$(arg control_height)" />
    </node>
</launch>